### PR TITLE
Fix datetime segment filter empty expression

### LINKED ISSUE #9762: Multiple areas of Mautic throw a 500 error
## Bug Description
I came across this issue when running the Cypress DWC tests and found it to also be reported in Emails (see #9646 #9755 and /https://forum.mautic.org/t/notices-trying-to-access-array-offset-on-value-of-type-null/18008).

I found it by trying to open a DWC item that was created but others find it with using certain email templates (with the legacy builder) as listed in the forum thread.

| Q   | A
| --- | ---
| Mautic version | 3.3.1
| PHP version | 7.4
| Browser | Chrome

### Steps to reproduce
1. Create dynamic content
2. Click to edit dynamic content
3. Notice a 503 error and log entry below
 
### Log errors

`mautic.NOTICE: PHP Notice - Trying to access array offset on value of type null - in file /var/www/html/app/bundles/EmailBundle/Views/FormTheme/Email/_emailform_dynamicContent_entry_filters_entry_filters_entry_widget.html.php - at line 14 {"mauticTemplateVars":{"app":"[object] (Symfony\\Bundle\\FrameworkBundle\\Templating\\GlobalVariables: {})","value":null,"attr":{"class":"form-control"},"form":"[object] (Symfony\\Component\\Form\\FormView: null)","id":"emailform_dynamicContent_0_filters_0_filters___name__","name":"__name__","full_name":"emailform[dynamicContent][0][filters][0][filters][__name__]","disabled":false,"label":false,"label_format":null,"multipart":false,"block_prefixes":["form","dynamic_content_filter_entry_filters","_emailform_dynamicContent_entry_filters_entry_filters_entry"],"unique_block_prefix":"_emailform_dynamicContent_entry_filters_entry_filters_entry","translation_domain":null,"cache_key":"_emailform_dynamicContent_entry_filters_entry_filters_entry_dynamic_content_filter_entry_filters","errors":"[object] (Symfony\\Component\\Form\\FormErrorIterator: )","valid":true,"data":null,"required":true,"size":null,"label_attr":[],"compound":true,"method":"POST","action":"","submitted":false,"mauticContent":"email","mauticTemplate":"MauticEmailBundle:FormTheme\\Email:_emailform_dynamicContent_entry_filters_entry_filters_entry_widget.html.php","view":"[object] (Mautic\\CoreBundle\\Templating\\Engine\\PhpEngine: {})"},"app":"[object] (Symfony\\Bundle\\FrameworkBundle\\Templating\\GlobalVariables: {})","value":null,"attr":{"class":"form-control"},"form":"[object] (Symfony\\Component\\Form\\FormView: null)","id":"emailform_dynamicContent_0_filters_0_filters___name__","name":"__name__","full_name":"emailform[dynamicContent][0][filters][0][filters][__name__]","disabled":false,"label":false,"label_format":null,"multipart":false,"block_prefixes":["form","dynamic_content_filter_entry_filters","_emailform_dynamicContent_entry_filters_entry_filters_entry"],"unique_block_prefix":"_emailform_dynamicContent_entry_filters_entry_filters_entry","translation_domain":null,"cache_key":"_emailform_dynamicContent_entry_filters_entry_filters_entry_dynamic_content_filter_entry_filters","errors":"[object] (Symfony\\Component\\Form\\FormErrorIterator: )","valid":true,"data":null,"required":true,"size":null,"label_attr":[],"compound":true,"method":"POST","action":"","submitted":false,"mauticContent":"email","mauticTemplate":"MauticEmailBundle:FormTheme\\Email:_emailform_dynamicContent_entry_filters_entry_filters_entry_widget.html.php","view":"[object] (Mautic\\CoreBundle\\Templating\\Engine\\PhpEngine: {})","isPrototype":true,"filterType":"","filterGroup":""} []
`

### PR DESCRIPTION
**Pull request by kuzmany**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:
If you use datettime segment filter and use empty, then edit page return 

```
Internal Server Error - PHP Notice - Trying to access array offset on value of type null 
/app/bundles/LeadBundle/Form/DataTransformer/FieldFilterTransformer.php:58 at
```

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Crate segment with datetime and choose empty expression
3. Save segment and try edit it
4. You should see error on dev enviroment

image

<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/10806"></a>

### CODE REVIEW

File: app/bundles/LeadBundle/Form/DataTransformer/FieldFilterTransformer.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public function transform($rawFilters)
                 $rawFilters[$k] = array_merge($this->default, $rawFilters[$k]);
             }
             if ('datetime' == $f['type']) {
-                if (in_array($f['filter'], $this->relativeDateStrings) or stristr($f['filter'][0], '-') or stristr($f['filter'][0], '+')) {
+                if (!$f['filter'] || in_array($f['filter'], $this->relativeDateStrings) or stristr($f['filter'][0], '-') or stristr($f['filter'][0], '+')) {
```
Comment (mmarton) [inline app/bundles/LeadBundle/Form/DataTransformer/FieldFilterTransformer.php]:
```suggestion
                if (!$f['filter'] || in_array($f['filter'], $this->relativeDateStrings) || stristr($f['filter'][0], '-') || stristr($f['filter'][0], '+')) {
```

`||` vs `or`  
Not affected by the PR but the mixed style within the condition hurts my eye and Mautic follows symfony conding style, so this should be `||` everywhere.

**Review by mmarton — DISMISSED** [review]
tested, and can confirm the issue ant the fix

File: app/bundles/LeadBundle/Form/DataTransformer/FieldFilterTransformer.php (hunk 1 above)
Comment (kuzmany) [reply app/bundles/LeadBundle/Form/DataTransformer/FieldFilterTransformer.php]: Yes, you're true. I forgot change old code. Thanks

**Comment by kuzmany** [comment]
Fixed by https://github.com/mautic/mautic/pull/8649

**Pull request closed without merging**